src/zmq/zmqpublishnotifier.cpp

```diff
@@ -195,3 +199,14 @@ bool CZMQPublishRawTransactionNotifier::NotifyTransaction(const CTransaction &tr
     ss << transaction;
     return SendMessage(MSG_RAWTX, &(*ss.begin()), ss.size());
 }
+
+bool CZMQPublishDecodedTransactionNotifier::NotifyTransaction(const CTransaction &transaction)
+{
+    uint256 hash = transaction.GetHash();
+    LogPrint(BCLog::ZMQ, "zmq: Publish decodedtx %s\n", hash.GetHex());
+    UniValue result(UniValue::VOBJ);
+    TxToUniv(transaction, uint256(), result, false);
```

Comment:
Have you tried `TxToJSON`?

Reply:
`TxToJSON` just calls `TxToUniv` internally, and also adds that extra `hashBlock` check, which is not strictly necessary imo. If we think that extra info would be useful though (confirmations, time, blocktime, blockhash), we can try to get the block hash beforehand and switch to using this method.